### PR TITLE
WELD-2501 Add reproducer, first attempt to fix this

### PR DESCRIPTION
**Pull request by manovotn**

### CODE REVIEW

File: impl/src/main/java/org/jboss/weld/util/reflection/Reflections.java [hunk 1]
```diff
@@ -465,14 +465,20 @@ public static Exception unwrapInvocationTargetException(InvocationTargetExceptio
         }
     }
 
-    public static Set<Class<?>> getInterfaceClosure(Class<?> clazz) {
+    public static Set<Class<?>> getInterfaceClosure(Class<?> clazz, Set<Class<?>> setOfInterfaces) {
```
Comment (manovotn) [inline impl/src/main/java/org/jboss/weld/util/reflection/Reflections.java]: We need to add a unit test to `org.jboss.weld.tests.unit.util.reflection.ReflectionsTest` to have a verification of this new behaviour.

Comment (mkouba) [reply impl/src/main/java/org/jboss/weld/util/reflection/Reflections.java]: Something like this should work:
```java
public static Set<Class<?>> getInterfaceClosure(Class<?> clazz) {
   Set<Class<?>> result = new HashSet<>();  
   for (Class<?> classToDiscover = clazz; classToDiscover != null; classToDiscover = classToDiscover.getSuperclass()) {
     addInterfaces(classToDiscover, result);
   }
   return result;
}
```

File: impl/src/main/java/org/jboss/weld/bean/proxy/InterceptedSubclassFactory.java [hunk 2]
```diff
@@ -238,6 +254,16 @@ void doReturn(CodeAttribute b, ClassMethod method) {
         }
     }
 
+    @Override
+    public Set<Class<?>> getAdditionalInterfaces() {
```
Comment (mkouba) [inline impl/src/main/java/org/jboss/weld/bean/proxy/InterceptedSubclassFactory.java]: I think we should not override the `getAdditionalInterfaces()` method but use the `interfacesToInspect` directly where needed. We don't want to extend the `additionalInterfaces` set as defined in `ProxyFactory` after all.

**Comment by manovotn** [comment]
Agreed via chat that the state of this PR is good to merge.

**Pull request merged**